content/en/docs/concepts/scheduling-eviction/assign-pod-node.md

```diff
@@ -378,6 +378,12 @@ The keys are used to look up values from the pod labels; those key-value labels
 (using `AND`) with the match restrictions defined using the `labelSelector` field. The combined
 filtering selects the set of existing pods that will be taken into Pod (anti)affinity calculation.
 
+{{< caution >}}
+It's not recommended to use `matchLabelKeys` with labels that might be updated directly on pods.
+Even if you edit the pod's label that is specified at `matchLabelKeys` **directly**, (that is, not via a deployment),
```

Comment:
nit: Deployment should have a capital D; also, we have more than one API for workload management (eg StatefulSet is a thing).